extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -152,56 +152,35 @@ export default class DebugPlugin extends BaseTelemetryPlugin {
-            [diagLog, diagLogProtoFns]
-          ] as any[]) {
-            for (const key of CoreUtils.objKeys(ext['__proto__'])) {
-              if (key.substring(0, 1) === '_') { continue; }
-              if (CoreUtils.isFunction(ext[key])) {
-                protoFns.push(key);
+          for (let i = 0; i < trackers.length; i++) {
+            let tracker = trackers[i];
+            for (const target of [propertiesExt, analyticsExt, ajaxDependencyExt, channelPluginExt, diagLog]) {
+              let val = InstrumentFunc(target, tracker, {
+                req: _self.preProcessItem(tracker) as any as () => InstrumentorHooksCallback,
+                rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
+              }, true);
+
+              if (!val) {
+                 val = InstrumentProto(target, tracker, {
+                  req: _self.preProcessItem(tracker) as any as () => InstrumentorHooksCallback,
+                  rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
+                });
+              }
+              if (val) {
+                debugBins[tracker] = new DebugBin(tracker, 0, debugBinParent, (i + 1) * 50);
+                break;
               }
             }
-            // special case for sender
-            if (ext.identifier === 'AppInsightsChannelPlugin' && CoreUtils.arrIndexOf(trackers, '_sender') !== -1) {
-              protoFns.push('_sender');
-            }
-          }
-
-          // 5. actually instrument all the functions
-          for (let i = 0; i < trackers.length; i++) {
-            const tracker = trackers[i];
-            let target;
-            if (CoreUtils.arrIndexOf(propertiesProtoFns, tracker) !== -1) { target = propertiesExt['__proto__'] }
-            else if (CoreUtils.arrIndexOf(analyticsProtoFns, tracker) !== -1) { target = analyticsExt['__proto__'] }
-            else if (CoreUtils.arrIndexOf(ajaxProtoFns, tracker) !== -1) { target = ajaxDependencyExt['__proto__'] }
-            else if (CoreUtils.arrIndexOf(diagLogProtoFns, tracker) !== -1) { target = diagLog['__proto__'] }
-            // special case for sender
-            else if (tracker === '_sender') { target = channelPluginExt }
-            else if (CoreUtils.arrIndexOf(channelProtoFns, tracker) !== -1) { target = channelPluginExt['__proto__'] }
-            else { continue; }
-            InstrumentFunc(target, tracker, {
-              req: _self.preProcessItem(tracker) as any as () => InstrumentorHooksCallback,
-              rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
-            });
-
-            debugBins[tracker] = new DebugBin(tracker, 0, debugBinParent, (i + 1) * 50);
           }
 
           // 6. append permanent style
           permStyleEl = document.createElement("style");
+          console.log(permStyle(prefix));
+          console.log(permStyle(prefix).substring(1000));
```

Comment:
remove

Reply:
+1 - The console is actually not defined by some browsers unless the dev console is open.